src/SideNav.ts

```diff
@@ -0,0 +1,180 @@
+import { LitElement, css, html } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import { classMap } from "lit/directives/class-map.js";
+
+import {
+  RouteTreeItem,
+  RouteTree,
+  RouteTreeItemType,
+  Route,
+  Routes,
+} from "./Routes";
+
+/**
+ * Main sidenav container element
+ * Renders tree of nav links based on provided routes
+ */
+@customElement("design-docs-sidenav")
+export class SideNavElement extends LitElement {
+  /**
+   * Flat array of routes
+   */
+  @property({ attribute: false, type: Array })
+  routes: Routes = [];
+
+  /**
+   * Tree-structured array of routes, will ideally be generated by this component,
+   * or possibly higher up to support title changes, TBD
+   */
+  @property({ attribute: false, type: Array })
+  routeTree: RouteTree = [];
+
+  @property({ attribute: true, type: Boolean })
+  searchable = false;
+
+  static styles = css`
+    :host {
+      font-family: var(--sidenav-font);
+    }
+    ul {
+      margin-block-start: var(--sidenav-item-spacing, 0.25em);
+      margin-block-end: var(--sidenav-item-spacing, 0.25em);
+    }
+    li:not(:first-child) {
+      margin-top: var(--sidenav-item-spacing, 0.25em);
+    }
+    li:not(:last-child) {
+      margin-bottom: var(--sidenav-item-spacing, 0.25em);
+    }
+    .hidden {
+      display: none;
+    }
+  `;
+
+  render() {
+    const searchClasses = { hidden: !this.searchable };
+    return html` <design-docs-sidenav-search
+        @search=${this.onSearch}
+        class=${classMap(searchClasses)}
+      ></design-docs-sidenav-search>
+      <nav>
+        <ul>
+          ${this.routeTree.map((item) => this.renderTreeItem(item))}
+        </ul>
+      </nav>`;
+  }
+
+  renderTreeItem(treeItem: RouteTreeItem) {
+    if (treeItem.type === RouteTreeItemType.GROUP) {
+      return html`<design-docs-sidenav-group label="${treeItem.label}">
+        ${treeItem.children?.map((item) =>
```

Comment:
Ideally we'd expect any `RouteTreeItem` with a type of `GROUP` to have a `children` prop defined, though I made it optional on the interface. In testing, excluding the `children` prop doesn't result in any list items being rendered, so I believe we're good. I know I read that lit treats `null` as an empty text node for rendering purposes, so it would seem it does the same for `undefined`